### PR TITLE
Opiate addict profession, retired drug war snippet, smoke shop only has legal drugs

### PR DESCRIPTION
**Pull request by ghost**
#### Summary
```SUMMARY: Content "Pillhead prof., retire drug war snippet, no illegal drugs in smoke shop"```

#### Purpose of change
There is no mention of the War on Drugs ending in the background lore document like is said in a snippet about it, and is contradicted by the crack pipe and ``drugs_analgesic`` item group description mentioning there are still illicit drugs. As such, smoke shops in the mall shouldn't stock illicit drugs. In addition, the lack of an opioid addict profession is strange, as all states comprimising New England are in the top 10 for deaths by overdose. 
#### Describe the solution
1. Removed the snippet for the WoD ending.
2. Added the Pillhead profession, who starts out addicted to opioids. -1 points.
3. Removed illegal drugs from the smoke_shop item group. Kept weed as every state in New England has a minimum a medical program.
#### Describe alternatives you've considered
Only adding the Pillhead profession, and keeping the snippet as a lorehack for the smoke shop storing otherwise illegal drugs. Would change crack pipe description for this solution.

#### Additional context
Source on Opioid Overdoses: https://www.drugabuse.gov/drugs-abuse/opioids/opioid-summaries-by-state
Map of marijuana legality: https://disa.com/map-of-marijuana-legality-by-state

### CODE REVIEW

File: data/json/professions.json [hunk 1]
```diff
@@ -921,6 +921,15 @@
     "items": { "both": [ "pants", "boxer_shorts" ], "male": [ "undershirt" ], "female": [ "camisole" ] },
     "addictions": [ { "intensity": 30, "type": "amphetamine" } ]
   },
+  {
+    "type": "profession",
+    "ident": "pillhead",
+    "name": "Pillhead",
+    "description": "After an accident in your youth, you got addicted to the opiates treating your pain. With the pharmacies shut down and dealers turned undead, satisfying your fix just got a lot more difficult.",
```
Comment (Night-Pryanik) [inline data/json/professions.json]:
```suggestion
    "description": "After an accident in your youth, you got addicted to the opiates treating your pain.  With the pharmacies shut down and dealers turned undead, satisfying your fix just got a lot more difficult.",
```

**Comment by kevingranade** [comment]
Needs styling

File: data/json/professions.json [hunk 2]
```diff
@@ -921,6 +921,15 @@
     "items": { "both": [ "pants", "boxer_shorts" ], "male": [ "undershirt" ], "female": [ "camisole" ] },
     "addictions": [ { "intensity": 30, "type": "amphetamine" } ]
   },
+  {
+    "type": "profession",
+    "ident": "pillhead",
+    "name": "Pillhead",
+    "description": "After an accident in your youth, you got addicted to the opiates treating your pain.  With the pharmacies shut down and dealers turned undead, satisfying your fix just got a lot more difficult.",
+    "points": -1,
+    "items": { "both": [ "sneakers","socks","jeans", "tshirt", "wristwatch", "oxycodone" ],"male": [ "boxer_shorts" ], "female": [ "panties", "bra" ] },
```
Comment (ampersand55) [inline data/json/professions.json]:
```suggestion
    "items": {
      "both": [ "sneakers", "socks", "jeans", "tshirt", "wristwatch", "oxycodone" ],
      "male": [ "boxer_shorts" ],
      "female": [ "panties", "bra" ]
    },
```

**Comment by LaVeyanFiend** [comment]
Linting/Astyling done

**Comment by LaVeyanFiend** [comment]
Jenkins rebuild

**Comment by LaVeyanFiend** [comment]
Guess Jenkins is still acting up, Mr. "failed to determine" 😛

**Comment by l29ah** [comment]
Okay, i suppose New England has a huge crack consumption epidemic among primary school students.

**Pull request merged**